File: CA/Projeto_1/python/edes_main.py
from des import *

plaintext = "Hello, World!"
key = "key"

sboxes=sbox_gen(random_gen(key,256))

#Padder
binary_plaintext = plaintext.encode()
padder = padding.PKCS7(512).padder() # 64 bytes = 512 bit
binary_plaintext= padder.update(binary_plaintext)
binary_plaintext += padder.finalize()

#Encrypt
encrypted_text = encrypt_edes(binary_plaintext, sboxes)

# Decrypt
decrypted_text = decrypt_edes(encrypted_text, sboxes)

#Unpadder
unpadder = padding.PKCS7(512).unpadder()
decrypted_text=unpadder.update(decrypted_text)
decrypted_text += unpadder.finalize()
decrypted_text = decrypted_text.decode()

print("Original:", plaintext)
print("Encrypted:", encrypted_text.hex())
print("Decrypted:", decrypted_text)

File: CA/Projeto_1/C++/EdesMain.cpp
#include <iostream>
#include <iomanip>
#include <vector>
#include <random>
#include <ctime>
#include <string>
#include <algorithm>
#include <stdint.h>
#include <openssl/crypto.h>
#include <openssl/evp.h>
#include "Sboxes.h"


int main(){
    
    std::string key = "key";
    std::string text = "Hello, World!";

    Sboxes box = Sboxes::sbox_gen(key);

    std::vector<unsigned char> bytes(text.begin(), text.end());

    //Encrypt
    bytes = addPaddingPKCS7(bytes);
    std::vector<unsigned char> encrypted_text = encryptEDES(bytes, box);

    //Decrypt
    std::vector<unsigned char> decrypted_text = decryptEDES(encrypted_text, box);
    decrypted_text = removePaddingPKCS7(decrypted_text);
    
    
    std::cout << "Original Text -> " + text << std::endl;

    std::cout << "Encrypted Text -> " ;
    for ( int i=0 ; i < encrypted_text.size(); i++){
        std::cout << std::hex << std::setfill('0') << std::setw(2) << (int)encrypted_text[i];
    }
    
    std::cout << std::endl << "Decrypted Text -> " ;
    for ( int i=0 ; i < decrypted_text.size(); i++){
        std::cout << decrypted_text[i] ;
    }
    std::cout << std::endl;
}